ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java

```diff
@@ -230,6 +231,14 @@ public DataStreamOutputRpc stream(ByteBuffer headerMessage) {
 
   @Override
   public DataStreamOutputRpc stream(ByteBuffer headerMessage, RoutingTable routingTable) {
+    if (routingTable != null && routingTable.getPrimary() != null) {
```

Comment:
Let's require `routingTable.getPrimary()` must be non-null.

Reply:
Thank you for the review. Updated based on the patch diff.